.github/workflows/deploy_feature_branch.yml

```diff
@@ -0,0 +1,134 @@
+name: "Reusable - Deploy Feature Branch to Dev"
+
+on:
+  workflow_call:
+    inputs:
+      terraform_workspace:
+        description: 'Terraform workspace'
+        required: true
+        type: string
+      role_session_name_suffix:
+        description: 'Suffix for the role session name'
+        required: true
+        type: string
+      deploy_fargate:
+        description: 'Deploy Fargate'
+        required: true
+        default: false
+        type: boolean
+      slack_notification_title:
+        description: 'Slack Notification Title'
+        required: false
+        type: string
+    secrets:
+      ORG_READ_ONLY_SSH_KEY:
+        required: true
+      ORG_GITHUB_PACKAGES_READ_ONLY_TOKEN:
+        required: true
+      CORE_ENG_DEPLOYMENTS_SLACK_WEBHOOK:
+        required: true
+    outputs:
+      name_prefix:
+        value: ${{ jobs.build_and_apply.outputs.name_prefix }}
+      api_endpoint:
+        value: ${{ jobs.build_and_apply.outputs.api_endpoint }}
+      deploy_fargate:
+        value: ${{ inputs.deploy_fargate }}
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  build_and_apply:
+    runs-on: ubuntu-latest
+    environment: dev
+    outputs:
+      name_prefix: ${{ steps.build_apply.outputs.name_prefix }}
+      api_endpoint: ${{ steps.build_apply.outputs.api_endpoint }}
+      deploy_fargate: ${{ inputs.deploy_fargate }}
+
+    steps:
+      - id: get-role-arn
+        uses: OpenSesame/gha-oidc-access/get-role-arn@v2
+        with:
+          domain: core
+          env: dev
+          ORG_READ_ONLY_SSH_KEY: ${{ secrets.ORG_READ_ONLY_SSH_KEY }}
+
+      - uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ steps.get-role-arn.outputs.role-arn }}
+          role-session-name: deployBranchDev-Run${{ inputs.role_session_name_suffix }}
+          aws-region: ${{ steps.get-role-arn.outputs.region }}
+
+      - name: Actions - Checkout Repo
+        uses: actions/checkout@v4
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20.x"
+          registry-url: "https://npm.pkg.github.com"
+
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: 1.6.3
+          terraform_wrapper: false
+
+      - name: Login to Amazon ECR
+        if: ${{ inputs.deploy_fargate }}
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Install IUM App Packages & Build App
+        if: ${{ inputs.deploy_fargate }}
+        run: |
+          cd lambdas/user-management-api
+          npm ci
+          npm run build
+
+      - name: Build And Apply Feature Branch to Dev
+        uses: opensesame/core-github-actions/build-apply@v1
```

Comment:
What would this workflow look like if we brought the steps directly into the workflow from this action? It would be nice to see from a workflow summary level where tasks like lint, build, apply terraform, run different test types are actually happening. Maybe that's why RevEng nested workflows instead of grouping things into actions...? How did the Simon team draw these lines?

Reply:
I did take a look at both reveng workflow as well as delivery team workflow. We can bring the steps directly into the action, but I kind of like hiding the steps in core github actions so it's not so noisy. but I'm open to change if that's the direction we want to go